ToDoList-Redux/Binders/ReduxFramework.swift

```diff
@@ -14,54 +14,72 @@ import CombineRex
 // MARK: - ACTIONS
 
 enum AppAction {
-    case task(TaskList.Action)
+    case appLifecycle(AppLifecycleAction)
+    case task(TaskAction)
 }
 
+enum TaskAction {
+    case add(String)
+    case remove(IndexSet)
+    case move(IndexSet, Int)
+    case toggle(String)
+}
 
 // MARK: - STATE
 
 struct AppState: Equatable {
     var appLifecycle: AppLifecycle
-    var task: TaskList.State
+    var tasks: [Task]
     
     static var empty: AppState {
-        .init(appLifecycle: .backgroundInactive, task: .empty)
+        .init(appLifecycle: .backgroundInactive, tasks: [])
     }
-    
+
     static var mock: AppState {
-        .init(appLifecycle: .backgroundInactive, task: .mock)
-    }
-    
-    enum AppLifecycle: Equatable {
-        case backgroundActive
-        case backgroundInactive
-        case foregroundActive
-        case foregroundInactive
+        .init(
+            appLifecycle: .backgroundInactive,
+            tasks: [
+                Task(title: "Implement UI", priority: .medium, completed: true),
+                Task(title: "Connect to Firebase", priority: .medium, completed: false),
+                Task(title: "????", priority: .high, completed: false),
+                Task(title: "PROFIT!!!", priority: .high, completed: false)
+            ]
+        )
     }
 }
 
 
 // MARK: - REDUCERS
 
+
 extension Reducer where ActionType == AppAction, StateType == AppState {
-    static let app = Reducer<TaskList.Action, TaskList.State>.task.lift(
-        action: \AppAction.task,
-        state: \AppState.task
-    )
+    static let app =
+        Reducer<TaskAction, [Task]>.task.lift(
+            action: \AppAction.task,
+            state: \AppState.tasks
+        ) <> Reducer<AppLifecycleAction, AppLifecycle>.lifecycle.lift(
```

Comment:
Great, you read my mind ! Was gonna ask about some examples of combining reducers with the `<>` operator. Nice.

Reply:
Something like
```
[
reducer1, reducer2
].reduce(.identity, <>)
```

also works. And it's fun to reduce reducers. :D